src/main/java/net/snowflake/client/core/arrow/tostringhelpers/ArrowObjectStringRepresentationBuilder.java

```diff
@@ -0,0 +1,30 @@
+package net.snowflake.client.core.arrow.tostringhelpers;
+
+import net.snowflake.client.core.SnowflakeJdbcInternalApi;
+import net.snowflake.client.jdbc.SnowflakeType;
+
+@SnowflakeJdbcInternalApi
+public class ArrowObjectStringRepresentationBuilder extends ArrowStringRepresentationBuilderBase {
+
+  public ArrowObjectStringRepresentationBuilder() {
+    super();
+    this.append('{');
+  }
+
+  public ArrowStringRepresentationBuilderBase appendKeyValue(
+      String key, String value, SnowflakeType valueType) {
+    addCommaIfNeeded();
+    this.appendQuoted(key).append(": ");
+    if (shouldQuoteValue(valueType)) {
+      return this.appendQuoted(value);
+    }
+
+    return this.append(value);
+  }
+
+  @Override
+  public String toString() {
+    this.append('}');
```

Comment:
If `toString()` method is used more than once, then we will append the closure everytime. Maybe it would be more feasible to append the postfix character within the returning `toString()` instead of this place, or even move it a step further and implement the logic within the base class using additional fields, e.g. `prefix` and `postfix`.

Reply:
For example, overloading the BuilderBase constructor with the prefix and postfix `ArrowStringRepresentationBuilderBase(char prefix, char postfix)` effectively moves its logic there and removes the `toString()` override from the child classes.